petastorm/fs_utils.py

```diff
@@ -33,6 +34,9 @@ def get_dataset_path(parsed_url):
         # s3/gs/gcs filesystem expects paths of the form `bucket/path`
         return parsed_url.netloc + parsed_url.path
 
+    if parsed_url.scheme.lower() in ['file'] and "Windows" in platform.system():
```

Comment:
It would be great if you could add a unit test verifying correctness of this function in `petastorm/tests/test_fs_utils.py`. You would need to mock the value returned by `platform.system()` and then I think the test logic should be straight forward.

If for some reason you can not do this, we'd be able to land this and I can help with the test later.